tutorials/understanding-ibc-denoms/index.md

```diff
@@ -5,201 +5,378 @@ order: 0
+
+<HighlightBox type="info">
 
-The `voucher` tokens introduced in the asset transfer are called IBC Denominations (IBC denom). The voucher tokens are the result of a token transfer using IBC from one blockchain to another. The format of the voucher token is:
+**So... why use a hash?**
 
-`ibc/DENOMHASH`.
+Hashing functions have many desirable properties that make them often used in cryptography, the property most useful in this discussion is that it always reduced to an output of fixed length (256 bits in case of SHA256), no matter the input.
 
-Imagine that you've received a new `ibc/` token on blockchain B where you initially held `samoleans` and `stake` token.
+Consider the following:
 
-Your balance now looks like:
+- The hash could contain paths that track the token on multiple hops from chain to chain.
+- This path could potentially be unbearably long when directly printing the path.
+- The Cosmos SDK has a 64-character limit on the denomination of the token.
 
-`1000000ibc/CDC4587874B85BEA4FCEC3CEA5A1195139799A1FEE711A07D972537E18FDA39D,100000000000samoleans,99999977256stake`
+Hence why a hash was preferred, more information on the design decisions can be found [here](https://ibc.cosmos.network/main/architecture/adr-001-coin-source-tracing.html).
 
-Just like `samoleans` or `stake`, `ibc/CDC458787...` is the denomination (denom) of the token received from IBC. After `ibc/CDC458787...` is a hash of the denom, the IBC port, and the channel.
+</HighlightBox>
 
-Why is `CDC458787...` a hash?
+The tradeoff of using a hash is that you cannot compute the input, given the output (hashing is an irreversible operation). Therefore the ICS20 module keeps a mapping of IBC denominations it has encountered to look up the orignal `path` and `base_denom`. Hence you are required to query a node to find out what the actual path and denomination is. This query is called the _denomtrace_.
+
+<HighlightBox type="docs">
+
+We see that in IBC denoms, there is a special meaning for the '/' character. We use it to parse port and channel identifiers from the base denom. Hence it was initially forbidden to use forward slashes in the base denomination of tokens.
+<br>
+However, due to a requirement from the Evmos chain, which uses forward slashes in contract-based denoms, support for base denominations containing '/' has been added. For more information, check the [ibc-go documentation](https://ibc.cosmos.network/main/migrations/support-denoms-with-slashes.html).
+
+</HighlightBox>
+
+## Practical example: `denom_trace`
+
+We can distinguish two cases of interacting with IBC denoms:
+
+1. to calculate the IBC denom for a given path
+2. to trace the path information and base denom when encountering an IBC denom
+
+The `transfer` IBC module exposes queries for both of these cases. In order to query, you'll have to interact with a node of a chain.
```

Comment:
@tmsdkeys  _"...you'll have to interact with **a node of a chain**."_

Is this use of "node" synonymous with _"a node of a _**network**_"_?

Reply:
Yes, if network is more consistent with the rest of the content that's fine to change